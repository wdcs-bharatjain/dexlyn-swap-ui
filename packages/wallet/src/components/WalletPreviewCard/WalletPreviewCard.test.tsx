import WalletPreviewCard from 'wallet/src/components/WalletPreviewCard/WalletPreviewCard'
import { SAMPLE_SEED_ADDRESS_1 } from 'wallet/src/test/fixtures'
import { render } from 'wallet/src/test/test-utils'

it('renders wallet preview card', () => {
  const tree = render(
    <WalletPreviewCard selected address={SAMPLE_SEED_ADDRESS_1} onSelect={(): null => null} />
  )
  expect(tree).toMatchSnapshot()
})
