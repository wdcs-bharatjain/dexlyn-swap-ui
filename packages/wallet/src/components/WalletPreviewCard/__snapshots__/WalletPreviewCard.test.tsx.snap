// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders wallet preview card 1`] = `
<View
  cancelable={true}
  disabled={false}
  focusable={true}
  hitSlop={[Function]}
  minPressDuration={0}
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#2222220D",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderBottomWidth": 1,
      "borderLeftColor": "#2222220D",
      "borderLeftWidth": 1,
      "borderRightColor": "#2222220D",
      "borderRightWidth": 1,
      "borderStyle": "solid",
      "borderTopColor": "#2222220D",
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "borderTopWidth": 1,
      "flexDirection": "column",
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "flexShrink": 1,
          "gap": 12,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "transparent",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 999999,
            "borderBottomRightRadius": 999999,
            "borderBottomWidth": 0,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 0,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 999999,
            "borderTopRightRadius": 999999,
            "borderTopWidth": 0,
            "flexDirection": "column",
            "position": "relative",
          }
        }
        testID="account-icon"
      >
        <svg
          height={36}
          viewBox="0 0 36 36"
          width={36}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            style={
              {
                "transformOrigin": "center center",
              }
            }
          >
            <circle
              cx={18}
              cy={18}
              fill="#00C3A01F"
              r={18}
            />
            <g
              transform="translate(6, 6) scale(0.5)"
            >
              <path
                clipRule="evenodd"
                d="M20.9454 0.0175773C22.4936 -0.167833 23.763 1.13917 23.763 2.72944V21.3625C23.763 22.9527 22.4993 24.2419 20.9405 24.2419L2.67549 24.2419C1.11665 24.2419 -0.16452 22.9469 0.0172307 21.3675C1.30306 10.1936 9.99222 1.32931 20.9454 0.0175773Z"
                fill="#00C3A0"
                fillRule="evenodd"
              />
              <path
                clipRule="evenodd"
                d="M27.0546 47.9824C25.5064 48.1678 24.237 46.8608 24.237 45.2706V26.6375C24.237 25.0473 25.5007 23.7581 27.0595 23.7581L45.3245 23.7581C46.8833 23.7581 48.1645 25.0531 47.9828 26.6325C46.6969 37.8064 38.0078 46.6707 27.0546 47.9824Z"
                fill="#00C3A0"
                fillRule="evenodd"
              />
            </g>
          </g>
        </svg>
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "flexShrink": 1,
            "gap": 0,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 12,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 4,
                "justifyContent": "center",
              }
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              maxFontSizeMultiplier={1.4}
              numberOfLines={1}
              style={
                {
                  "color": "#222222",
                  "flexShrink": 1,
                  "fontFamily": "Basel-Book",
                  "fontSize": 18,
                  "lineHeight": 24,
                }
              }
              suppressHighlighting={true}
              testID="address-display/name/0x​82D5...3Fa6"
            >
              0x​82D5...3Fa6
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="20"
        bbWidth="20"
        fill="none"
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        strokeWidth={8}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "color": "#FC72FF",
              "height": 20,
              "width": 20,
            },
            {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        tintColor="#FC72FF"
        vbHeight={16}
        vbWidth={16}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "strokeWidth",
            ]
          }
          strokeWidth="8"
        >
          <RNSVGPath
            d="M13.3334 4L6.00008 11.3333L2.66675 8"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
